src/main/java/racingcar/domain/NumberOfPower.java

```diff
@@ -0,0 +1,11 @@
+package racingcar.domain;
+
+import java.util.List;
+
+public abstract class NumberOfPower {
+
+    protected static final int MIN_NUMBER_OF_POWER = 0;
+    protected static final int MAX_NUMBER_OF_POWER = 9;
+
+    protected abstract List<Integer> generateNumbersOfPower(int size);
+}
```

Comment:
이 클래스는 인터페이스로 선언해도 되는데 추상 클래스로 선언한 이유가 있나요?

```suggestion
public interface NumberOfPower {

    int MIN_NUMBER_OF_POWER = 0;
    int MAX_NUMBER_OF_POWER = 9;

    List<Integer> generateNumbersOfPower(int size);
}
```

Reply:
이 경우엔 인터페이스가 더 적절하다고 생각되어 리팩토링 했습니다.
### 인터페이스가 적절한 이유?
1. `generateNumbersOfPower` 이외에 다른 메서드가 필요하지 않다.(확장 가능성이 낮다)
2. 구현 객체마다(`RandomNumberGenerator`, `CustomNumberOfPower`) 같은 동작을 보장하기 위한 용도로 사용된다.
3. 클래스 간 연간 관계가 없다.(추상 클래스의 용도)

### 추상클래스로 구현했던 이유?
- 확장성 때문입니다. 자식 클래스에 따라, 부모 클래스에서 정의하지 않은 메서드가 포함될 수도 있을 것이라고 생각했습니다.
- 그러나, 이번 미션에서는 확장 가능성이 큰 의미가 없는 것 같습니다.

Reply:
제가 느끼기엔 인터페이스와 추상클래스가 별 차이는 없다고 생각합니다 ㅎ
둘다 향후 확장을 위한 규격서 라고 봐용

차이가 있다면 추상클래스는 상태와 템플릿메서드를 미리 세팅해놓을 수 있는 부분이라고 생각합니다

그래서 굳이 상태와 공통 메서드가 필요없는 요 경우에선 추상클래스가 부적절해 보였습니다!